src/back/spv/writer.rs

```diff
@@ -87,12 +92,166 @@ impl Writer {
+                id
+            }
+        }
+    }
+
+    fn get_global_variable_id(
+        &mut self,
+        arena: &crate::Arena<crate::Type>,
+        global_arena: &crate::Arena<crate::GlobalVariable>,
+        handle: crate::Handle<crate::GlobalVariable>,
+    ) -> Word {
+        match self.lookup_global_variable.lookup_id(handle) {
+            Some(word) => word,
+            None => {
+                let global_variable = &global_arena[handle];
+                let (instruction, id) = self.parse_global_variable(arena, global_variable, handle);
+                instruction.to_words(&mut self.logical_layout.declarations);
+                id
+            }
+        }
+    }
+
+    fn get_function_type(
+        &mut self,
+        ty: Option<crate::Handle<crate::Type>>,
+        arena: &crate::Arena<crate::Type>,
+    ) -> Word {
+        match ty {
+            Some(handle) => self.get_type_id(arena, handle),
+            None => match self.void_type {
+                Some(id) => id,
+                None => {
+                    let id = self.generate_id();
+                    self.void_type = Some(id);
+                    self.instruction_type_void(id)
+                        .to_words(&mut self.logical_layout.declarations);
+                    id
+                }
+            },
+        }
+    }
+
+    fn get_pointer_id(
+        &mut self,
+        arena: &crate::Arena<crate::Type>,
+        handle: crate::Handle<crate::Type>,
+        class: spirv::StorageClass,
+    ) -> Word {
+        let ty = &arena[handle];
+        let type_id = self.get_type_id(arena, handle);
+        match ty.inner {
+            crate::TypeInner::Pointer { .. } => type_id,
+            _ => {
+                let pointer_id = self.generate_id();
+                let instruction = self.instruction_type_pointer(pointer_id, class, type_id);
+                instruction.to_words(&mut self.logical_layout.declarations);
+
+                /* TODO
+                    Not able to lookup Pointer, because there is no Handle in the IR for it.
+                    Idea would be to not have any handles at all in the lookups, so we aren't bound
+                    to the IR. We can then insert, like here runtime values to the lookups
+                */
+                // self.lookup_type.insert(pointer_id, global_variable.ty);
+                pointer_id
+            }
+        }
+    }
+
+    fn find_scalar_handle(
+        &self,
+        arena: &crate::Arena<crate::Type>,
+        kind: crate::ScalarKind,
+        width: u8,
+    ) -> crate::Handle<crate::Type> {
+        let mut scalar_handle = None;
+        for (handle, ty) in arena.iter() {
+            match ty.inner {
+                crate::TypeInner::Scalar {
+                    kind: _kind,
+                    width: _width,
+                } => {
+                    if kind == _kind && width == _width {
+                        scalar_handle = Some(handle);
```

Comment:
why not just `return handle`?

Reply:
I could, but the goal of this function would be determining if a scalar exists in the IR, and ultimately create the scalar if it is not present in the IR. This would solve #85. To fix this, I have to refactor the whole Lookup architecture I created in this first version.

I experimented with a new lookup architecture in boids. I think best option would be to not solve it now, but to do this _**before**_ I am going to work on the boids example and PR this separately.